src/hooks/usePatchStatusSelect.ts

```diff
@@ -28,58 +83,96 @@ export const usePatchStatusSelect = (
         newState[id] = true;
       }
     } else {
+      // Enter this condition when a parent checkbox is clicked.
+      // If every task is already checked, uncheck them. Otherwise, check them.
+      const nextCheckedState = !id.every((taskId) => selectedTasks[taskId]);
       id.forEach((taskId) => {
-        if (newState[taskId]) {
-          newState[taskId] = false;
-        } else {
-          newState[taskId] = true;
-        }
+        newState[taskId] = nextCheckedState;
       });
     }
-
-    setSelectedTasks(newState);
+    dispatch({ type: "setSelectedTasks", data: newState });
   };
 
-  // Iterate through PatchBuildVariants and determine if a task should be selected or not
-  // Based on if the task status correlates with the validStatus filter
+  // Determine if a task is a selected based on the filter terms and available tasks
+  const prevPatchBuildVariants = usePrevious(patchBuildVariants);
+  const prevPatchStatusFilterTerm = usePrevious(patchStatusFilterTerm);
+  const prevBaseStatusFilterTerm = usePrevious(baseStatusFilterTerm);
   useEffect(() => {
-    if (patchBuildVariants) {
-      let tempSelectedTasks = selectedTasks;
-      patchBuildVariants.forEach((patchBuildVariant) => {
-        patchBuildVariant.tasks.forEach((task) => {
-          if (validStatus.includes(task.status)) {
-            tempSelectedTasks = addTaskToSelectedTasks(
-              task.id,
-              tempSelectedTasks
-            );
-          } else {
-            tempSelectedTasks = removeTaskFromSelectedTasks(
-              task.id,
-              tempSelectedTasks
-            );
-          }
+    const filterTermOrPatchTasksChanged =
+      patchBuildVariants !== prevPatchBuildVariants ||
+      patchStatusFilterTerm !== prevPatchStatusFilterTerm ||
+      baseStatusFilterTerm !== prevBaseStatusFilterTerm;
+    if (filterTermOrPatchTasksChanged) {
+      if (window.Worker && webWorker) {
+        webWorker.postMessage({
+          patchBuildVariants,
+          patchStatusFilterTerm,
+          baseStatusFilterTerm,
+          selectedTasks,
         });
-      });
-      setSelectedTasks(tempSelectedTasks);
+      } else {
+        // fallback in case web workers are not available.
+        // This code reflects logic in public/web_worker/patchBuildVariantReduce.js
+        // Iterate through PatchBuildVariants and determine if a task should be
+        // selected or not based on if the task status correlates with the 2 filters.
+        // if 1 of the 2 filters is empty, ignore the empty filter
```

Comment:
The following logic exists in the web worker but are unable to share the code since the web worker file is being hosted from the `public` directory. I think it could make sense to include it in our `src` folder by installing additional node modules and/or by ejecting CRA if a) we increasingly find ourselves using web workers to solve problems b) we find other significant benefits from ejecting CRA

Reply:
I am all for ejecting CRA . I think if this is the approach we would like to take and something we plan on doing more of, CRA is limiting us at this point. There are other advantages i think ejecting CRA would provide such as giving us more ability to customize jest. And also allow us to update and customize our build tooling.